test/lib/util.js

```diff
@@ -27,6 +27,14 @@ function concatJson(resolve, reject) {
   };
 }
 
+function delaysJson(ms, json) {
+  return function (req, res) {
+    setTimeout(function () {
```

Comment:
we do not want an actual timeout; rather, we want to fake time with a timer library somewhere and have timeout values of 5 seconds or so.

Reply:
agreed.  (it's just always a pain to mock http timeouts!)  will see if i can tackle the rest of this with a real-slow test suite, then try to mock out the timers.